cocotb/decorators.py

```diff
@@ -180,18 +180,19 @@ def __nonzero__(self):
             otherwise return true"""
         return not self._finished
 
-    # Needs `yield from` syntax to implement this correctly.
     # Once 2.7 is dropped, this can be run unconditionally
     if sys.version_info >= (3, 3):
         exec_(textwrap.dedent("""
         def __await__(self):
-            if self._natively_awaitable:
```

Comment:
This flag doesn't seem to be used any more?

Reply:
It's not, but I thought it might be worth leaving in anyway, just in case it becomes useful for debugging to know what type of coroutine is being dealt with. If you don't think that is a particularly good justification, I'll remove it.

Reply:
If you feel it's useful then let's leave it in: you're probably the main user of it and know what helps during debugging :-)